client/pdo/client/controller/commands/create.py

```diff
@@ -132,53 +130,33 @@ def command_create(state, bindings, pargs) :
     logger.info('Loaded contract code for %s', contract_class)
 
     # ---------- set up the enclave clients ----------
-    enclaveclients = []
-    enclave_names = state.get(['Service', 'EnclaveServiceNames'], [])
-    if len(enclave_names) > 0: #use the database to get the list of enclaves for the contract
-        logger.info('Using eservice database to look up service URL for the contract enclave')
-        try:
-            for name in enclave_names:
-                enclaveclients.append(db.get_client_by_name(name))
-        except Exception as e:
-            raise Exception('Unable to get the eservice clients using the eservice database: %s', str(e))
-
-    else:
-        eservice_urls = state.get(['Service', 'EnclaveServiceURLs'], [])
-        if len(eservice_urls) == 0 :
-            raise Exception('no enclave services specified')
-        try:
-            for url in eservice_urls :
-                enclaveclients.append(EnclaveServiceClient(url))
-        except Exception as e :
-            raise Exception('unable to contact enclave services; {0}'.format(str(e)))
+    eservice_clients = get_eservice_list(state, options.eservice_group)
+    if len(eservice_clients) == 0 :
+        raise Exception('unable to locate enclave services in the group %s', options.eservice_group)
 
+    preferred_eservice_client = get_eservice(state, eservice_group=options.eservice_group)
 
```

Comment:
the current implementation of get_esrvice sets a preferred eservice for every contract, which means that it is not  possible to randomize the choice of enclave for different updates. It might be better not to set any preferred enclave if the client does not explicitly say one. This will be useful if we want to test replication with the shell.

Reply:
yes it is. just set preferred='random'

Reply:
I guess my point is that if the client does not want any preferred enclave, the get_eservice still sets a preferred enclave (chosen at random). In summary, all udpates run on a single enclave (either chosen explicitly or chosen once at random)

Reply:
no... the string "random" says choose at will on every execution.

Reply:
i am afraid i dont understand. For instance, the shell-tests.psh does not set any preferred enclave, but the pdo files created all have a preferred enclave entry which I would not have expected. If the field is there in the pdo file, I imagine that this enclave is used for all updates

Reply:
1) the eservice groups except for the default one all specify a preferred enclave already.
2) the preferred enclave is picked when the contract is created, either from the group or from the command line option (which may be random); the preferred enclave is NOT the only enclave, it is simply the one that is used by default.
3) the send command allows any enclave to be specified for execution including "random" to pick an enclave at random